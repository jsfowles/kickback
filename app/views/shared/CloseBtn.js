import React from 'react';
import {
  TouchableHighlight,
  Image,
  StyleSheet,
} from 'react-native';

const CloseBtn = ({ onPress, color }) => (
  <TouchableHighlight
    style={ styles.btn }
    onPress={ onPress }
    underlayColor='transparent'
  >
    <Image
      source={ require('image!close') }
      style={{ tintColor: color }}
    />
  </TouchableHighlight>
);

CloseBtn.propTypes = {
  onPress: React.PropTypes.func.isRequired,
  color: React.PropTypes.string.isRequired,
};

CloseBtn.defaultProps = {
  color: '#ffffff',
};

const styles = StyleSheet.create({
  btn: {
    position: 'absolute',
    top: 16,
    right: 16,
  },
});

export default CloseBtn;
