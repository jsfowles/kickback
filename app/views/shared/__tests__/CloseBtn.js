/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import CloseBtn from '../CloseBtn';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <CloseBtn onPress={ () => null } />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with a color passed in', () => {
  const tree = renderer.create(
    <CloseBtn color='#000000' onPress={ () => null } />
  ).toJSON();

  expect(tree.children[0].props.style.tintColor).toBe('#000000');
});

it('renders correctly with no color passed in', () => {
  const tree = renderer.create(
    <CloseBtn onPress={ () => null } />
  ).toJSON();

  expect(tree.children[0].props.style.tintColor).toBe('#ffffff');
});
