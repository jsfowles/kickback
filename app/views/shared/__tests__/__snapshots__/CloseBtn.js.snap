exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent"
      },
      Object {
        "position": "absolute",
        "right": 16,
        "top": 16
      }
    ]
  }
  testID={undefined}>
  <Image
    source={
      Object {
        "__packager_asset": true,
        "deprecated": true,
        "height": 100,
        "path": "/full/path/to/something.png",
        "uri": "icon",
        "width": 100
      }
    }
    style={
      Object {
        "tintColor": "#ffffff"
      }
    } />
</View>
`;
